src/Workleap.Extensions.Mongo.Tests/MongoIndexerTests.Inheritance.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.Extensions.DependencyInjection;
+using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Driver;
+using Workleap.Extensions.Mongo.Indexing;
+
+namespace Workleap.Extensions.Mongo.Tests;
+
+public partial class MongoIndexerTests
+{
+    [Fact]
+    public async Task UpdateIndexesAsync_Ignores_Automatically_Inherited_Index_Provider()
+    {
+        async Task AssertIndexes<TDocument>()
+            where TDocument : IMongoDocument
+        {
+            using var indexCursor = await this.Services.GetRequiredService<IMongoCollection<TDocument>>().Indexes.ListAsync();
+            var indexNames = await indexCursor.ToAsyncEnumerable().Select(x => x["name"].AsString).ToArrayAsync();
+
+            Assert.Equal(2, indexNames.Length);
+            Assert.Contains("_id_", indexNames);
+            Assert.Contains("name_7a2f315f44b4b7813c1db6c66b1ae173", indexNames);
+        }
+
+        var documentTypes = new[] { typeof(BaseTestDocumentA) };
+        await this.Services.GetRequiredService<IMongoIndexer>().UpdateIndexesAsync(documentTypes);
+
+        await AssertIndexes<BaseTestDocumentA>();
+        await AssertIndexes<ChildTestDocumentB>();
+        await AssertIndexes<ChildTestDocumentC>();
+    }
+
+    [BsonKnownTypes(typeof(ChildTestDocumentB), typeof(ChildTestDocumentC))]
+    [MongoCollection("inheritedIndexProviders", IndexProviderType = typeof(BaseTestDocumentAIndexProvider))]
+    private class BaseTestDocumentA : MongoDocument
```

Comment:
This was our user's setup.
- A base document class that provides an index provider type, and has two derived types marked as `BsonKnownTypes`
- The two derived types don't have to be decorated with `MongoCollection` nor have to provide an index provider.

The derived types here benefit from the indexes declared by their base type.